clang/tools/dpct/cmake/dpct.cmake

```diff
@@ -78,3 +78,48 @@ endmacro()
 # Always set SYCL_HAS_FP16 to true to assume SYCL device to support float16
 message("dpct.cmake: SYCL_HAS_FP16 is set true by default.")
 set(SYCL_HAS_FP16 TRUE)
+
+# Replaces cuda_compile with it's equivalent SYCL repr
+macro(DPCT_COMPILE generated_files)
```

Comment:
According to the definition of cuda_compile
cuda_compile(<generated_files> <file>... [STATIC | SHARED | MODULE]
             [OPTIONS ...])
Returns a list of generated files from the input source files to be used with [add_library()](https://cmake.org/cmake/help/latest/command/add_library.html#command:add_library) or [add_executable()](https://cmake.org/cmake/help/latest/command/add_executable.html#command:add_executable).

Reply:
`dpct_compile` use the create static or dynamic libs based on the type passed to `cuda_compile`. But I see that is not the same behaviour as `cuda_compile`.

I have modified the `dpct.cmake` to return a list of object file path as `generated_files`. Please take a look.

Reply:
I'm OK to current modification.